### PR TITLE
registration feature added

### PR DESCRIPTION
**Pull request by MuthuKumari-P**
- Customer Registration Details are validated
-  Customer Registration Details are added to the DB
- Extra validation are added to the frontend

### CODE REVIEW

File: src/main/java/in/muthukumari/dao/ExistDAO.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+	/**
+	 * This method used to check if the usename is exists or not
+	 * 
+	 * @param userName
+	 * @return
+	 * @throws DBException
+	 */
+	public static boolean isExistUserName(String userName) throws DBException {
+		Connection con = null;
```
Comment (selva08sankari) [inline src/main/java/in/muthukumari/dao/ExistDAO.java:24]: declare con,pst,rs as class variable

File: src/main/java/in/muthukumari/dao/ExistDAO.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	}
+
+	/**
+	 * This method used to check if the email is exists or not
+	 * 
+	 * @param email
+	 * @return
+	 * @throws DBException
+	 */
+	public static boolean isExistEmail(String email) throws DBException {
```
Comment (selva08sankari) [inline src/main/java/in/muthukumari/dao/ExistDAO.java:93]: why you are checking each input exist or not in different method? check in one method

File: src/main/java/in/muthukumari/validator/BankDetailValidator.java [hunk 3]
```diff
@@ -111,20 +117,25 @@ public static boolean isValidMobileNumber(CustomerBankDetail customer) {
...
+			branchAndIfscList = BankDAO.getBranchNameAndIfscList(customer.getBankName());
+			if (branchAndIfscList.containsKey(customer.getBranchName())) {
+				String ifsc = branchAndIfscList.get(customer.getBranchName());
+				if (ifsc.equals(customer.getIfscCode())) {
+					isValidIfsc = true;
+				} else {
+					isValidIfsc = false;
+				}
 			}
+		} catch (DBException e) {
```
Comment (selva08sankari) [inline src/main/java/in/muthukumari/validator/BankDetailValidator.java:135]: validator only throw validator exception not others

File: src/main/java/in/muthukumari/validator/BankDetailValidator.java [hunk 4]
```diff
@@ -46,16 +49,18 @@ public static boolean isValidAccountNumber(CustomerBankDetail customer) throws C
...
+		Set<String> bankNameList = null;
 		boolean isValid = false;
-		if (bankNameList.contains(customer.getBankName())) {
-			isValid = true;
+		try {
+			bankNameList = BankDAO.getBankNameList();
+			if (bankNameList.contains(customer.getBankName())) {
+				isValid = true;
+			}
+		} catch (DBException e) {
```
Comment (selva08sankari) [inline src/main/java/in/muthukumari/validator/BankDetailValidator.java:62]: validator only throw validator exception

File: src/main/java/in/muthukumari/validator/CustomerRepeatedValidator.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	public static boolean isRepeatedEmail(String email) throws CustomerRepeatedException {
+		boolean isRepeatedEmail = true;
+		try {
+			boolean isExists = ExistDAO.isExistEmail(email);
+			if (isExists) {
+				isRepeatedEmail = false;
+			}
+
+		} catch (DBException e) {
+			throw new CustomerRepeatedException("The email id is already given");
```
Comment (selva08sankari) [inline src/main/java/in/muthukumari/validator/CustomerRepeatedValidator.java]: here also throw validator exception

**Review by selva08sankari — CHANGES_REQUESTED** [review]

File: src/main/java/in/muthukumari/validator/BankDetailValidator.java (hunk 3 above)
Comment (MuthuKumari-P) [reply src/main/java/in/muthukumari/validator/BankDetailValidator.java:135]: I have changed

File: src/main/java/in/muthukumari/validator/BankDetailValidator.java (hunk 4 above)
Comment (MuthuKumari-P) [reply src/main/java/in/muthukumari/validator/BankDetailValidator.java:62]: I have changed

File: src/main/java/in/muthukumari/validator/CustomerRepeatedValidator.java (hunk 5 above)
Comment (MuthuKumari-P) [reply src/main/java/in/muthukumari/validator/CustomerRepeatedValidator.java]: I have changed this

File: src/main/java/in/muthukumari/dao/ExistDAO.java (hunk 1 above)
Comment (MuthuKumari-P) [reply src/main/java/in/muthukumari/dao/ExistDAO.java:24]: The variables scope is end at the method end so I declared it as local variable

File: src/main/java/in/muthukumari/dao/ExistDAO.java (hunk 2 above)
Comment (MuthuKumari-P) [reply src/main/java/in/muthukumari/dao/ExistDAO.java:93]: Because After validation the values are stored in the database

**Pull request merged**